frontend/src/views/WheresMyGene/components/HeatMap/utils.ts

```diff
@@ -360,15 +342,18 @@ export function dataToChartFormat(
 
       const scaledMeanExpression = (meanExpression - min) / oldRange;
 
-      const geneIndex = genes.findIndex((gene) => gene === geneName);
+      const geneIndex = genes.findIndex((gene) => gene.name === geneName);
 
-      const cellTypeIndex = cellTypes.findIndex(
-        (cellType) => cellType.id === dataPoint.id
+      const cellTypeIndex = cellTypeSummaries.findIndex(
+        (cellTypeSummary) => cellTypeSummary.id === dataPoint.id
       );
 
+      const id = `${dataPoint.id}-${geneName}`;
```

Comment:
this `id` will be used for echarts rendering optimization later, when we have time to figure out logics to only update affected dots when cell types and genes change